### PR TITLE
Add zero-cross for Intents

### PR DESCRIPTION
**Pull request by kbrizzle**
Enables crossing zero in delta update methods (for intent trading).

Adds testing around invariant updates to support this feature.

### CODE REVIEW

File: packages/core/test/unit/market/Market.test.ts [hunk 1]
```diff
@@ -407,6 +409,19 @@ async function deposit(market: Market, amount: BigNumber, account: SignerWithAdd
     )
 }
 
+async function getOrderProcessingEvents(
```
Comment (EdNoepel) [inline packages/core/test/unit/market/Market.test.ts:412]: Where is this used?  Could the one from `fees.test.ts` have been moved into _common_ and reused?

**Review by EdNoepel — APPROVED** [review]
Looks good.  As far as integration testing, should something in `happyPath.test.ts` exercise a zero-cross?

File: packages/core/test/unit/market/Market.test.ts (hunk 1 above)
Comment (kbrizzle) [reply packages/core/test/unit/market/Market.test.ts:412]: I use it heavily while debugging so I thought I'd add it in here instead of copy-pasting it every time.

**Comment by kbrizzle** [comment]
I added a follow up task to add integration tests for zero-crosses.
